clients/src/test/java/org/apache/kafka/common/network/NioEchoServer.java

```diff
@@ -60,14 +60,8 @@
     public enum MetricType {
         TOTAL, RATE, AVG, MAX;
 
-        private final String metricNameSuffix;
-
-        private MetricType() {
-            metricNameSuffix = "-" + name().toLowerCase(Locale.ROOT);
-        }
-
         public String metricNameSuffix() {
-            return metricNameSuffix;
+            return "-" + name().toLowerCase(Locale.ROOT);
```

Comment:
This replaces a single allocation during construction with allocations on every method call.